src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -137,10 +154,51 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
   closePopover() {
     this.setState({ ...this.state, isPopoverOpen: false });
   }
+  getFilteredDefaultDataSource(defaultDataSource: string | null): string | null {
+    if (!defaultDataSource) return defaultDataSource;
+    const { dataSourceFilter } = this.props;
+    // check if the defaultDataSouce can be filtered out
+    const filteredDataSource = dataSourceFilter
+      ? [{ id: defaultDataSource }].filter(dataSourceFilter)
```

Comment:
data source filter received a data source object, we should pass in the complete data source object fetched from saved object   `dataSourceFilter?: (dataSource: any) => boolean;`